src/btree.ts

```diff
@@ -0,0 +1,135 @@
+import { BPTreeNode, MemoryPointer, compareBytes } from "./node";
+
+// taken from `buffer.go`
+interface ReadWriteSeekTruncater {
+	write(buffer: Uint8Array): Promise<number>;
+	seek(offset: number, whence: "start" | "current" | "end"): Promise<number>;
+	read(buffer: Uint8Array, offset: number): Promise<number>;
```

Comment:
The go pattern of writing to a buffer is a bit of an antipattern in javascript. Since the http request allocates a buffer anyways, writing to a buffer will result in a copy which defeats the benefit that passing the output buffer offers.

In other words, the read function is really just a `read(offset: number): Promise<Uint8Array>`, which is what the RangeResolver provides: https://github.com/kevmo314/appendable/blob/main/src/resolver.ts#L16-L23